extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -274,6 +325,18 @@ void beans(BuildProducer<AdditionalBeanBuildItem> beanProducer) {
             beanProducer.produce(AdditionalBeanBuildItem.unremovableOf(CamelMainProducers.class));
         }
 
+        @BuildStep(onlyIf = Flags.MainEnabled.class)
```

Comment:
maybe add some doc explaining why it is needed

Reply:
Haha, you've got me. It's a result of copy and paste. Let me figure out if I actually want that.

Reply:
maybe add also some lines about the reason of adding unremoveable beans